src/library/scala/collection/Map.scala

```diff
@@ -374,7 +363,7 @@ object MapOps {
   */
 @SerialVersionUID(3L)
 object Map extends MapFactory.Delegate[Map](immutable.Map) {
-  private val DefaultSentinel: AnyRef = new AnyRef
+  private val DefaultSentinel: () => AnyRef = () => new AnyRef
```

Comment:
what's the rationale for changing this to be a function that allocates an object each time, rather than just being a static object?

Reply:
When an object is passed as by-name argument to `getOrElse`, a function is allocated. The change here avoids that function allocation, because there's an special case in the compiler for `getOrElse(foo())` where `foo` is a `Function0`. However, doing `new AnyRef` as the result of the function is unnecessary allocation, we can just return `null` or any other static / constant value. I'll change that.

Reply:
(possibly not `null`, as that's theoretically a valid result from `Map#get`, as much as Scala discourages the use of `null`?)

Reply:
Right...